src/services/measure.service.js

```diff
@@ -42,4 +45,72 @@ const remove = async args => {
   return baseRemove(args, 'Measure');
 };
 
-module.exports = { create, searchById, remove, update };
+/**
+ * takes a measureReport and a set of required data with which to calculate the measure and
+ * creates new documents for the measureReport and requirements in the appropriate collections
+ * @param {*} args the args object passed in by the user
+ * @param {*} req the request object passed in by the user
+ * @returns a transaction-response bundle
+ */
+const submitData = async (args, { req }) => {
+  if (req.body.resourceType !== 'Parameters') {
+    throw new ServerError(null, {
+      statusCode: 400,
+      issue: [
+        {
+          severity: 'error',
+          code: 'BadRequest',
+          details: {
+            text: `Expected 'resourceType: Parameters'. Received 'type: ${req.body.resourceType}'.`
+          }
+        }
+      ]
+    });
+  }
+  if (!req.body.parameter) {
+    throw new ServerError(null, {
+      statusCode: 400,
+      issue: [
+        {
+          severity: 'error',
+          code: 'BadRequest',
+          details: {
+            text: `Unreadable or empty entity for attribute 'parameter'. Received: ${req.body.parameter}`
+          }
+        }
+      ]
+    });
+  }
+
+  const tb = new TransactionBundle();
+  const parameters = req.body.parameter;
+
+  let containsMeasureReport = false;
```

Comment:
I slightly prefer doing a short-circuit here rather than always going through the whole loop just to fail after if a `measureReport` parameter isn't there.

Suggested approach would be to do a `.filter` on the `parameters` resource for something where `param.name === 'measureReport'` and throw an error if that resulting array's length is `!== 1`. This will also properly check the case where there are more than 1 `measureReport` param, which is not allowed by the operation. The current behavior only checks if one doesn't exist at all.